File: bindings/php/FieldProgramUnaryOp.php
<?php
namespace Stencil;

final class FieldProgramUnaryOp {
    const Neg = 0;
    const Abs = 1;
    const Sqrt = 2;
    const Log = 3;
    const Sin = 4;
    const Cos = 5;
    const Acos = 6;
    const VecX = 7;
    const VecY = 8;
    const VecZ = 9;
    const Length = 10;
    const Normalize = 11;

    public static function name(int $value): string {
        return match ($value) {
            self::Neg => 'Neg',
            self::Abs => 'Abs',
            self::Sqrt => 'Sqrt',
            self::Log => 'Log',
            self::Sin => 'Sin',
            self::Cos => 'Cos',
            self::Acos => 'Acos',
            self::VecX => 'VecX',
            self::VecY => 'VecY',
            self::VecZ => 'VecZ',
            self::Length => 'Length',
            self::Normalize => 'Normalize',
            default => "Unknown({$value})",
        };
    }
}

File: bindings/kotlin/src/main/kotlin/at/schem/nucleation/FieldProgramUnaryOp.kt
package at.schem.nucleation

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface FieldProgramUnaryOpLib: Library {
}
/** Unary field-program operations (see `crate::sdf::UnaryOp`).
*/
enum class FieldProgramUnaryOp {
    Neg,
    Abs,
    Sqrt,
    Log,
    Sin,
    Cos,
    Acos,
    VecX,
    VecY,
    VecZ,
    Length,
    Normalize;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<FieldProgramUnaryOpLib> = FieldProgramUnaryOpLib::class.java
        internal val lib: FieldProgramUnaryOpLib = Native.load("nucleation", libClass)
        fun fromNative(native: Int): FieldProgramUnaryOp {
            return FieldProgramUnaryOp.entries[native]
        }

        fun default(): FieldProgramUnaryOp {
            return Neg
        }
    }
}
